internal/pkg/instrumentation/bpf/database/sql/probe_test.go

```diff
@@ -18,7 +18,59 @@ import (
 	"go.opentelemetry.io/auto/internal/pkg/instrumentation/utils"
 )
 
+func BenchmarkProcessFn(b *testing.B) {
+	tests := []struct {
+		name  string
+		query string
+	}{
+		{
+			name:  "no query (baseline)",
+			query: "",
+		},
+		{
+			name:  "simple query",
+			query: "SELECT * FROM customers",
+		},
+		{
+			name:  "medium query",
+			query: "SELECT * FROM customers WHERE first_name='Mike' AND last_name IN ('Santa', 'Banana')",
+		},
+		{
+			name:  "hard query",
+			query: "WITH (SELECT last_name FROM customers WHERE first_name='Mike' AND country='North Pole') AS test_table SELECT * FROM test_table WHERE first_name='Mike' AND last_name IN ('Santa', 'Banana')",
+		},
+	}
+
+	start := time.Unix(0, time.Now().UnixNano()) // No wall clock.
+	end := start.Add(1 * time.Second)
+
+	startOffset := utils.TimeToBootOffset(start)
+	endOffset := utils.TimeToBootOffset(end)
+
+	traceID := trace.TraceID{1}
+	spanID := trace.SpanID{1}
+
+	for _, t := range tests {
+		b.Run(t.name, func(b *testing.B) {
+			var byteQuery [256]byte
+			copy(byteQuery[:], []byte(t.query))
+			b.ResetTimer()
```

Comment:
So we can see memory benchmark.

```suggestion
			b.ReportAllocs()
			b.ResetTimer()
```